raiden/utils/__init__.py

```diff
@@ -101,11 +102,32 @@ def address_decoder(addr: str) -> typing.Address:
     return addr
 
 
+def address_checksum_and_decode(addr: str) -> typing.Address:
+    """ Accepts a string address and turns it into binary.
+
+        Makes sure that the string address provided starts is 0x prefixed and
+        checksummed according to EIP55 specification
+    """
+    if addr[:2] != '0x':
+        raise InvalidAddress('All provided addresses must be 0x prefixed')
+
+    if not is_checksum_address(addr):
+        raise InvalidAddress('Provided addresses must be EIP55 checksummed')
+
+    addr = unhexlify(addr[2:])
+    assert len(addr) in (20, 0)
+    return addr
+
+
 def address_encoder(address: typing.Address) -> str:
     assert len(address) in (20, 0)
     return '0x' + hexlify(address).decode()
 
 
+def address_encoder_and_checksum(address: typing.Address) -> str:
+    return to_checksum_address(address_encoder(address))
```

Comment:
we don't really this utility, `to_checksum_address` should be sufficient to convert the binary representation to string and do the checksum:

```
>>> from raiden.tests.utils import factories
>>> factories.HOP1
b'w\x95,\xe8<\xa3\xca\xd9\xf7\xad\xcf\xab\xed\xa8[\xd2\xf1\xf5 \x08'
>>> address.to_checksum_address(factories.HOP1)
'0x77952Ce83Ca3cad9F7AdcFabeDA85Bd2F1f52008'
```